StratisCore.UI/src/app/wallet/cold-staking/components/overview/wallet/wallet.component.ts

```diff
@@ -37,6 +50,48 @@ export class ColdStakingWalletComponent {
         this.amountFormatted = this._amount.toLocaleString();
     }
 
+    ngOnInit() {
+        this.wallet = this.walletService.wallet();
+        this.buildStakingForm();
+    }
+
     unusedAddressClicked = () => this.onGetFirstUnusedAddress.emit(this);
     withdrawClicked = () => this.onWithdraw.emit(this);
+
+    private buildStakingForm(): void {
+        this.stakingForm = this.fb.group({
+            'walletPassword': ['', Validators.required]
+        });
+    }
+
+    public startStaking(): void {
+        const walletData = {
+          name: this.globalService.getWalletName(),
+          password: this.stakingForm.get('walletPassword').value
+        };
+    
+        this.stakingForm.patchValue({walletPassword: ''});
+    
+        if (this.stakingService.startStaking(walletData)) {
```

Comment:
This is a promise<boolean> thats returned not boolean so this will always be true